csrc/ops/arith.cpp

```diff
@@ -1289,31 +1296,17 @@ TensorView* reductionOp(
       "\n  Domain: ",
       tv->domain()->toString());
 
-  TORCH_CHECK(axes.size() > 0, "No reduction axis specified");
+  TORCH_CHECK(!axes.empty(), "No reduction axis specified");
 
   auto tv_root = TensorDomain::noReductions(tv->getMaybeRFactorDomain());
-  const int ndims = tv_root.size();
+  const auto ndims = tv_root.size();
 
   // PyTorch allows reduction of 0-dim tensors
   if (ndims == 0) {
     return reductionOpZeroDimTensor(tv);
   }
 
-  std::vector<unsigned int> uint_axes;
-  for (int axis : axes) {
-    if (axis < 0) {
-      axis += ndims;
-    }
-
-    TORCH_CHECK(
-        axis >= 0 && axis < ndims,
-        "Reduction on invalid axis, received: ",
-        axis,
-        " however tensor view only has ",
-        ndims,
-        " non-reduction dims.");
-    uint_axes.push_back((unsigned int)axis);
-  }
+  std::vector<unsigned int> uint_axes = canonicalizeAxes(axes, ndims);
```

Comment:
Should we make the type of everything here, including the parameter of`reductionOp` and `canonicalizeAxes`, `newForMma`, `newForReduction`, and `fusedMultiplySum` (and maybe more) use the same type `std::vector<int64_t>`?